src/Symfony/Component/Messenger/Bridge/Redis/Transport/RedisReceiver.php

```diff
@@ -47,6 +48,15 @@ public function get(): iterable
         $redisEnvelope = json_decode($message['data']['message'] ?? '', true);
 
         if (null === $redisEnvelope) {
+            if (null !== $message['id']) {
+                try {
+                    $this->connection->ack($message['id']);
+                } catch (TransportException $e) {
+                }
+
+                return $this->get();
+            }
```

Comment:
Forgotten to mention here. I don't have a better idea, as there is no possibility to change connection setting `deleteAfterAck` from outside the connection, and there is no possibility to perform `xack` directly from this class. If anyone has an idea how to make it more elegant, feel free to suggest how to do it.

Reply:
I'm not sure if we should catch the `TransportException` here. Would that not hide an error if connection get lost? /cc @chalasr 

Also maybe it would be better ro call `$this->connection->reject` as we are doing nothing with that message.

If somebody has `deleteAfterAck=false&deleteAfterReject=true` that message would stay in the stream for other consumers, which I think would make more sense then using ack here?

Reply:
> maybe it would be better ro call $this->connection->reject as we are doing nothing with that message

what about this proposal @jvmanji ?